StRoot/StBTofUtil/StBTofSimResParams.h

```diff
@@ -20,80 +21,99 @@
 // class tofSimResParams_st;
 
 class StBTofSimResParams : public StMaker {
-public:
+	public:
 
 		StBTofSimResParams() {}
 		~StBTofSimResParams() {}
 
-double average_timeres_tof(){return mAverageTimeResTof;}
+		double average_timeres_tof(){return mAverageTimeResTof;}
 
-	/**
-	* Calculates the average resolution across all 38 tubes (discounts inactive tubes)
-	* then returns a single vertex resolution (in ps) for use in embedding w/ vpdStart
-	*/
-	double timeres_tof(uint itray, uint imodule, uint icell) {
-		double result = 8.5e-11;
-		if ( itray > 120 || imodule > 32 || icell > 6 )
-			return result;
+		/**
+		 * Calculates the average resolution across all 38 tubes (discounts inactive tubes)
+		 * then returns a single vertex resolution (in ps) for use in embedding w/ vpdStart
+		 */
+		double timeres_tof(uint itray, uint imodule, uint icell) 
+		{
+			double result = 8.5e-11;
+			if ( itray > 120 || imodule > 32 || icell > 6 )
+				return result;
 
-		return params[ itray ][ imodule * 6 + icell ];
-
-	}
-
-	//! Loads Vpd Sim Params from database
-
-	void loadParams(const int date = 20160913, const int time = 175725, const char* Default_time = "2016-09-13 17:57:25")
-	{
-
-		St_db_Maker *dbMk = dynamic_cast<St_db_Maker*>( GetChain()->GetMakerInheritsFrom("St_db_Maker") );
+			return params[ itray ][ imodule * 6 + icell ];
+		}
 
-		TDataSet *DB = GetDataBase("Calibrations/tof/tofSimResParams");
+		//! Loads BTOF Sim Params from database
+		//void loadParams(const int date = 20160913, const int time = 175725, const char* Default_time = "2016-09-13 17:57:25")
+		void loadParams(const int runNumber = 20076002)
+		{
+			
+			const int yearNumber = runNumber/1000000 - 1 + 2000;
+			const int date       = yearNumber*1000*10 + 101; // year + 01-01(Date)
 
-		if (!DB) {
-			LOG_WARN << "No data set found, creating new St_db_Maker... with date/time" << date << "/" << time << endm;
-			dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
+			St_db_Maker *dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
 			dbMk->SetDebug();
-			dbMk->SetDateTime(date,time); //! event or run start time, set to your liking
+			dbMk->SetDateTime(date,1);
+			cout<<"cout for test, data: "<<date<<endl;
```

Comment:
This clearly for debugging only. Remove this cout

Reply:
Will do in next update